x-pack/platform/plugins/shared/task_manager/server/lib/create_managed_configuration.ts

```diff
@@ -164,18 +151,28 @@ function createPollIntervalScan(logger: Logger, startingPollInterval: number) {
             );
             newPollInterval = previousPollInterval;
           }
+
+          // If the task claim strategy is mget, increase the poll interval if the the avg used capacity over 15s is less than 25%.
+          avgTmUtilization = getAverageUtilization(tmUtilization, window);
+          if (claimStrategy === CLAIM_STRATEGY_MGET && newPollInterval < DEFAULT_POLL_INTERVAL) {
+            updatedForCapacity = true;
+            if (avgTmUtilization < 25) {
+              newPollInterval = DEFAULT_POLL_INTERVAL;
+            } else {
+              // If the the used capacity is greater than or equal to 25% reset the polling interval.
+              newPollInterval = startingPollInterval;
+            }
+          }
         }
       }
-
       if (newPollInterval !== previousPollInterval) {
         if (previousPollInterval !== INTERVAL_AFTER_BLOCK_EXCEPTION) {
-          logger.debug(
-            `Poll interval configuration changing from ${previousPollInterval} to ${newPollInterval} after seeing ${errorCount} "too many request" and/or "execute [inline] script" and/or "cluster_block_exception" error(s).`
-          );
-        }
-        if (previousPollInterval === startingPollInterval) {
           logger.warn(
-            `Poll interval configuration is temporarily increased after Elasticsearch returned ${errorCount} "too many request" and/or "execute [inline] script" and/or "cluster_block_exception" error(s).`
+            `Poll interval configuration changing from ${previousPollInterval} to ${newPollInterval} after ${
+              updatedForCapacity
+                ? `a change in the average task load: ${avgTmUtilization}.`
+                : `seeing ${errorCount} "too many request" and/or "execute [inline] script" error(s).`
```

Comment:
I'm curious why `and/or "cluster_block_exception" error(s)` isn't in the new message.  I'm wondering if Ersin merged some of the cluster-block-checking code between when this PR was created, and now.  And then got missed in the conflicts.

Reply:
oh you're prob right, sorry about that! There were some messy merge conflicts on this pr

Reply:
Resolved in this commit https://github.com/elastic/kibana/pull/200260/commits/5db8c7aaece3a7fbb632873597e69b389070061e

Reply:
Perhaps @ersin-erdal should take a peek as well?